ray-operator/test/e2e/rayservice_ha_test.go

```diff
@@ -197,3 +198,83 @@ func TestRayServiceZeroDowntimeUpgrade(t *testing.T) {
 		"python", "/locust-runner/locust_runner.py", "-f", "/locustfile/locustfile.py", "--host", "http://test-rayservice-serve-svc:8000",
 	})
 }
+
+func TestRayServiceGCSFaultTolerance(t *testing.T) {
+	rayserviceYamlFile := "testdata/ray-service.ft.yaml"
+	locustYamlFile := "testdata/locust-cluster.const-rate.yaml"
+
+	test := With(t)
+	g := NewWithT(t)
+
+	// Create a namespace
+	namespace := test.NewTestNamespace()
+
+	// Create a ConfigMap with Locust runner script
+	configMapAC := newConfigMap(namespace.Name, "locust-runner-script", files(test, "locust_runner.py"))
+	configMap, err := test.Client().Core().CoreV1().ConfigMaps(namespace.Name).Apply(test.Ctx(), configMapAC, TestApplyOptions)
+	g.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created ConfigMap %s/%s successfully", configMap.Namespace, configMap.Name)
+
+	// Create the RayService for testing
+	KubectlApplyYAML(test, rayserviceYamlFile, namespace.Name)
+	rayService, err := GetRayService(test, namespace.Name, "test-rayservice")
+	g.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created RayService %s/%s successfully", rayService.Namespace, rayService.Name)
+
+	test.T().Logf("Waiting for RayService %s/%s to running", rayService.Namespace, rayService.Name)
+	g.Eventually(RayService(test, rayService.Namespace, rayService.Name), TestTimeoutMedium).
+		Should(WithTransform(RayServiceStatus, Equal(rayv1.Running)))
+
+	g.Expect(GetRayService(test, rayService.Namespace, rayService.Name)).To(WithTransform(RayServicesNumEndPoints, Equal(int32(1))))
```

Comment:
You should use `Eventually`. Currently, when RayService is `Running`, it does not indicate that the endpoint has already been attached to the Kubernetes service.

Reply:
OK, no problem